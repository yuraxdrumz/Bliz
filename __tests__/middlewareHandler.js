// // handler for express middlewares with next...
// async function midHandler(Promise, req, res, arr){
//   function next(resolve, reject, ...args){
//     if(args.length > 0) return reject(args[0])
//     return resolve()
//   }
//
//   for(let item of arr){
//     await new Promise((resolve, reject)=>{
//       item(req,res,next.bind(this,resolve, reject))
//     })
//   }
// }
//
// export default midHandler
import midHandler from '../src/middlewareHandler'
describe('Middleware handler suite', ()=>{
  const fn1 = jest.fn((req,res,next)=>next())
  const fn2 = jest.fn((req,res,next)=>next())
  const fn3 = jest.fn((req,res,next)=>next())
  const req = {}
  const res = {}
  const arr = [
    fn1,
    fn2,
    fn3
  ]
  test('should iterate on functions array received and inject next to each', async ()=>{
    expect(await midHandler(Promise, req, res, arr)).not.toBeDefined()
    expect(fn1.mock.calls[0]).toMatchSnapshot()
    expect(fn2.mock.calls[0]).toMatchSnapshot()
    expect(fn3.mock.calls[0]).toMatchSnapshot()
  })
})