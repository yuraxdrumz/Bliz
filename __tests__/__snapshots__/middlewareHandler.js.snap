// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middleware handler suite should iterate on functions array received and inject next to each 1`] = `
Array [
  Object {},
  Object {},
  [Function],
]
`;

exports[`Middleware handler suite should iterate on functions array received and inject next to each 2`] = `
Array [
  Object {},
  Object {},
  [Function],
]
`;

exports[`Middleware handler suite should iterate on functions array received and inject next to each 3`] = `
Array [
  Object {},
  Object {},
  [Function],
]
`;
